### PR TITLE
[plugin.video.teliaplay-se] v1.2.3+matrix.1

### PR DESCRIPTION
**Pull request by Ikosse**
### Description
In order to avoid creating a new PR any time the graphql hashes changes, the hashes are now obtained from my dedicated repository: https://github.com/Ikosse/teliaplay-graphql-hashes
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.teliaplay-se/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+__pycache__/
```
Comment (basrieter) [inline plugin.video.teliaplay-se/.gitignore]: Don't submit the .gitignore file

File: plugin.video.teliaplay-se/resources/lib/api.py [hunk 2]
```diff
@@ -60,6 +40,25 @@ def __init__(self, userdata):
             self.token_data = None
         self.web_utils = WebUtils()
 
+    @property
+    def graphql_hashes(self):
+        request = {
+            "GET": {
+                "scheme": "https",
+                "host": "raw.githubusercontent.com",
```
Comment (basrieter) [inline plugin.video.teliaplay-se/resources/lib/api.py]: @enen92 Do we mind this being on his github user content space?

Comment (enen92) [reply plugin.video.teliaplay-se/resources/lib/api.py]: Yeah :/

Comment (Ikosse) [reply plugin.video.teliaplay-se/resources/lib/api.py]: Would be possible to include if I made it optional and with a big disclaimer?

Comment (enen92) [reply plugin.video.teliaplay-se/resources/lib/api.py]: Does it really changes that frequently that can't be handled via repo submissions/updates?

Comment (Ikosse) [reply plugin.video.teliaplay-se/resources/lib/api.py]: My impression is that it varies; sometimes they become obsolete after ~2 weeks and sometimes they have worked for more than a month. It also seem to vary depending on the request; i.e., some menu entries stops working more frequently than others. I suppose the issue is that they change both frequently and unpredictably, and having a separate repository allows me to respond quicker.

I guess the optimal solution would be to generate the hashes on the fly, or send the full graphql request (if that's even is possible). I've tried figuring out how the browser does it, but not with much luck as my current knowledge of graphql is, let's say, limited... My guess is that they are retrieved/generated through some javascript routine? I would be most grateful if you (or anyone) could provide me with some pointers in this direction. :)

Comment (enen92) [reply plugin.video.teliaplay-se/resources/lib/api.py]: Maybe some better option is to look into their hypothetical mobile applications and check how they do it there. I doubt they issue mobile application updates on a 2 week basis

Comment (Ikosse) [reply plugin.video.teliaplay-se/resources/lib/api.py]: Great, will do, thank you! I will keep updating the hashes through PRs then.

Comment (basrieter) [reply plugin.video.teliaplay-se/resources/lib/api.py]: I you update the PR, I can re-review it.

Comment (Ikosse) [reply plugin.video.teliaplay-se/resources/lib/api.py]: Done; hopefully everything looks alright now.

**Review by basrieter — APPROVED** [review]

**Pull request merged**